pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -134,8 +164,32 @@ pub mod pallet {
 			domain: Domain,
 			hook_address: [u8; 20],
 		},
+
+		/// Message recovery was executed.
+		MessageRecoveryExecuted {
+			domain: Domain,
+			proof: Proof,
+			router_id: T::RouterId,
+		},
 	}
 
+	// TODO(cdamian): Add migration to clear this storage.
+	// /// Storage for domain routers.
+	// ///
+	// /// This can only be set by an admin.
+	// #[pallet::storage]
+	// #[pallet::getter(fn domain_routers)]
+	// pub type DomainRouters<T: Config> = StorageMap<_, Blake2_128Concat, Domain,
+	// T::Router>;
+
+	/// Storage for routers specific for a domain.
+	///
+	/// This can only be set by an admin.
+	#[pallet::storage]
+	#[pallet::getter(fn routers)]
+	pub type Routers<T: Config> =
+		StorageMap<_, Blake2_128Concat, Domain, BoundedVec<T::RouterId, T::MaxRouterCount>>;
```

Comment:
Ok, regarding this, because `RouterId` already contains the `Domain` information, the `set_domain_routers()` could be `set_routers()`.

Later, you use `routers.for_domain(domain)` to get the correct router list for a domain, but you should not store a domain per router. Basically because the gateway does not know if that domain is supported for that router

Reply:
@cdamian referring to the other thread ^^

Reply:
OK, lemme try something on a different branch to see if I can wrap my mind around this.

Reply:
Let's discuss more [here](https://github.com/centrifuge/centrifuge-chain/pull/1962/files)

Reply:
Nice! As far as I could see, you got the idea perfectly! 🙌🏻